### PR TITLE
Add lang/volume/speak key for :speak function to follow robot-interface 

### PR DESCRIPTION
**Pull request by k-okada**
you need to run `roslaunch naoqi_apps speech.launch ` to use `:wait` key.

### CODE REVIEW

**Review by Affonso-Gui — COMMENTED** [review]
Both `:set-` and `:speak` methods are permanent setters. I was kind of expecting to see a one-off change for the `:speak` key arguments (do we have a "/get_language" service?). This would also simplify interactions between both methods:
```lisp
(send *ri* :speak "Hello" :volume 0.5) ;; volume 0.5
(send *ri* :set-volume 0.1)
(send *ri* :speak "Hello" :volume 0.5)  ;; volume 0.1
```

**Comment by k-okada** [comment]
@Affonso-Gui you mean 
```
(send *ri* :speak "Hello" :volume 0.5) ;; volume 0.5
(send *ri* :set-volume 0.1)
(send *ri* :speak "Hello" :volume 0.5)  ;; volume 0.5
(send *ri* :speak "Hello")  ;; volume 0.1
```
???

**Comment by Affonso-Gui** [comment]
Yes, I believe that this should be the expected behavior. However, it is ultimately a matter of preference.
@kochigami

**Comment by kochigami** [comment]
Thank you very much for adding new features for Naoqi robots.
However, I’m very sorry, I can’t follow the discussion…
I would like to know when it would be useful to make this change.

For me, the way below is enough when I want to change the volume in the middle of an utterance.

```
(send *ri* :set-volume 0.4)
(send *ri* :speak "Hello") ;; volume 0.4

(send *ri* :set-volume 0.9)
(send *ri* :speak "Hello") ;; volume 0.9
```

**Comment by k-okada** [comment]
@kochigami thanks for comments, this change added `:lang` `:volume` key for `:speak` methods, becuase [robot-interface.l](https://github.com/jsk-ros-pkg/jsk_pr2eus/blob/4700cf84db91d3b8d6cb17a540c29add5fbfd656/pr2eus/robot-interface.l#L1304) have these arguments, and range of `:volume` is `0` to `1.0`

Thus, other robots, like pr2/fetch usesrs write code like `(send *ri* :speak "hello" :volume 0.8)` , and this PR support same code on pepper/nao robot.

NaoQi's `set-volume` uses `0` to `100` for volume range, and we do not change the `:set-volume` function, becuase this function is unique to NaoQi system

File: jsk_naoqi_robot/naoqieus/naoqi-interface.l [hunk 1]
```diff
@@ -328,15 +344,18 @@
...
    ()
    (let ((ret (instance nao_interaction_msgs::GetAudioMasterVolumeRequest :init))
-	 res)
+	 res volume)
      (ros::wait-for-service (format nil "~A/naoqi_driver/get_volume" group-namespace))
      (setq res (ros::service-call (format nil "~A/naoqi_driver/get_volume" group-namespace) ret))
-     (send res :master_volume :data)))
+     (setq volume (send res :master_volume :data))
+     (setq master-volume (/ volume 100.0))
+     vol))
```
Comment (kochigami) [inline jsk_naoqi_robot/naoqieus/naoqi-interface.l:358]: I think `vol` should be `volume`.

`send *ri* :get-master-volume` =>

```
/opt/ros/melodic/share/euslisp/jskeus/eus/Linux64/bin/irteusgl 0 error: unbound variable vol in (let ((ret (instance nao_interaction_msgs::getaudiomastervolumerequest :init)) res volume) (ros::wait-for-service (format nil "~A/naoqi_
driver/get_volume" group-namespace)) (setq res (ros::service-call (format nil "~A/naoqi_driver/get_volume" group-namespace) ret)) (setq volume (send res :master_volume :data)) (setq master-volume (/ volume 100.0)) vol)
```

Comment (kochigami) [reply jsk_naoqi_robot/naoqieus/naoqi-interface.l:358]: After modifying `vol` to `volume`, I confirmed this PR works well (Ubuntu 18.04, ROS melodic).

```
roslaunch jsk_pepper_startup jsk_pepper_startup.launch network_interface:=enp4s0 launch_joy:=false (this branch)
roslaunch naoqi_apps speech.launch (kochigami-develop)
```
```
roscd peppereus
roseus pepper-inteface.l
pepper-init t

2.irteusgl$ send *ri* :set-language "English"
t
3.irteusgl$ send *ri* :get-language
"English"
4.irteusgl$ send *ri* :set-master-volume 50
 #<nao_interaction_msgs::setaudiomastervolumeresponse #X55c4e7bd63f0>
5.irteusgl$ send *ri* :get-master-volume
50
6.irteusgl$ send *ri* :speak "Hello, Hello, Hello, Hello, Hello" :lang "Japanese" :wait t :volume 0.4
[ INFO] [1660552895.025133938]: subscribing speech_status
[ INFO] [1660552895.524033720]: subscribing speech_status
[ INFO] [1660552896.024239547]: subscribing speech_status
[ INFO] [1660552896.524247881]: subscribing speech_status
[ INFO] [1660552897.024213482]: subscribing speech_status
[ INFO] [1660552897.524022335]: subscribing speech_status
[ INFO] [1660552898.024220278]: subscribing speech_status
[ INFO] [1660552898.524135728]: subscribing speech_status
[ INFO] [1660552899.024391846]: subscribing speech_status
[ INFO] [1660552899.524035537]: subscribing speech_status
nil
7.irteusgl$ send *ri* :get-language
"Japanese"
8.irteusgl$ send *ri* :get-master-volume
40
9.irteusgl$ send *ri* :speak "Hello, Hello, Hello, Hello, Hello" :wait t :volume 0.4
[ INFO] [1660552929.352141912]: subscribing speech_status
[ INFO] [1660552929.851521853]: subscribing speech_status
[ INFO] [1660552930.351535601]: subscribing speech_status
[ INFO] [1660552930.851412365]: subscribing speech_status
[ INFO] [1660552931.351527093]: subscribing speech_status
[ INFO] [1660552931.851525856]: subscribing speech_status
[ INFO] [1660552932.351486681]: subscribing speech_status
[ INFO] [1660552932.851489413]: subscribing speech_status
[ INFO] [1660552933.351495158]: subscribing speech_status
nil
10.irteusgl$ send *ri* :speak "Hello, Hello, Hello, Hello, Hello" :wait t :volume 0.4 :lang "English"
[ INFO] [1660552947.863459217]: subscribing speech_status
[ INFO] [1660552948.362732784]: subscribing speech_status
```

**Review by kochigami — CHANGES_REQUESTED** [review]
In https://github.com/jsk-ros-pkg/jsk_robot/pull/1535/files#diff-970f29b01103b1f7fa143a89027401a28be8a290d8f597e8e1ceaa59ab51f266R358,
`vol` should be `volume`.

File: jsk_naoqi_robot/naoqieus/naoqi-interface.l (hunk 1 above)
Comment (k-okada) [reply jsk_naoqi_robot/naoqieus/naoqi-interface.l]: @kochigami thank you fixed

**Review by Affonso-Gui — CHANGES_REQUESTED** [review]
This implementation still have a few problems:
- The initial `master-volume` value of 0.5 is never enforced
- Setting the volume in line 255 re-sets the `master-volume` slot value, so the original volume is never restored
- The program cannot keep track of changes done outside of the setters and getters, e.g. service call, direct slot setting, other eus interfaces...

I have created a PR in https://github.com/k-okada/jsk_robot/pull/74 to address these and other problems.
The PR code:
- Prioritizes getters over slot variables to access the current state (the service calls seemed to be fast enough)
- Forces the :wait setting when attempting to run one-off operations. This is required because the naoqi system will adjust the settings immediately upon request.
- Adds some synonyms when setting the language

**Comment by kochigami** [comment]
> Forces the :wait setting when attempting to run one-off operations. This is required because the naoqi system will adjust the settings immediately upon request.

As for this, when we call `(send *ri* :speak "Hello, Hello, Hello, Hello, Hello" :lang "English" :volume 0.3)`, 
without `:wait` setting, 
the volume of the first "Hello" is 0.3, but that of other "Hello" is 0.7. 

https://user-images.githubusercontent.com/7259671/185333351-66d95170-b380-447d-8736-ce8e11ad550e.MP4

**Review by Affonso-Gui — APPROVED** [review]

**Pull request merged**